modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/utils/sortItems.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {IOrderable} from './types';
+
+/**
+ * sorts the provided items array according to the itemsOrder comma-separated list of ids.
+ * If array contains items not included in the list of ids, then those are appended after
+ * Example:
+ * 		items = [ {id: 1}, {id: 4}, {id: 2}, {id: 3} ]
+ * 		itemsOrder = "2, 3, 1"
+ * 		output is [ {id: 2}, {id: 3}, {id: 1}, {id: 4} ]
+ * Optionally, not included items can be sorted by creation date
+ *
+ * @param items {IOrderable[]}
+ * @param itemsOrder {string}
+ * @param useCreationDate {boolean}
+ * @returns {Array}
+ */
+export default function sortItems(
+	items: IOrderable[],
+	itemsOrder: string,
+	useCreationDate: boolean = false
```

Comment:
Don't we want to sort by creation date by default?

Reply:
yes, we want it sorted. 

The nit here is: if data already [comes sorted](https://github.com/liferay-frontend/liferay-portal/pull/4173/commits/86a5c00ed6130eb31eabbf05a9de686d32e6b84c#diff-03068c08f64ffdd4ab7b85cada1d396fa420791dbf11f7dfe7cd6c5928b6a062R140) (which is possible for all cases except filters), then we can't avoid this last step. I turned it on only when data comes from [various sources](https://github.com/liferay-frontend/liferay-portal/pull/4173/commits/86a5c00ed6130eb31eabbf05a9de686d32e6b84c#diff-5d5125c41012dc4363b655295b5996c3c2df1eb521120ffa4c46038925e9cb70R489-R512) (filters)

Your comment made me see a place where I did not apply this: [fields](https://github.com/liferay-frontend/liferay-portal/pull/4173/commits/86a5c00ed6130eb31eabbf05a9de686d32e6b84c#diff-04631558821004cdb02882757dbe1f9e4bfdec979f6a4aa8ddec19bb94084e97R362). Will change it now!